### PR TITLE
settingUp.md: Include additional instructions for Mac users on parts that differ from Windows. #4631

### LINKED ISSUE #4631: settingUp.md: Include additional instructions for Mac users on parts that differ from Windows.
There are some parts in the setting up guide that is only applicable to Windows, i.e. the step is a bit different for Mac users. 

One example is that the "Preferences" of Eclipse in Mac is found in the "Eclipse" menu in the top left hand corner instead of the "Windows" menu in the menu bar.

### PR DESCRIPTION
**Pull request by Gisonrg**
Fixes #4631

### CODE REVIEW

File: devdocs/settingUp.md [hunk 1]
```diff
@@ -179,7 +180,19 @@ this section.
   `runtests.bat  C:\appengine-java-sdk-1.9.27  C:\teammates -testnames component-tests,sequential-ui-tests,parallel-ui-tests`
 
 **On OS-X:**
-TBD
+* Build the project in Eclipse (`Project -> Clean`).
+* Start the dev server in Eclipse.
+* Open a Terminal window, navigate to the project folder and run the `runtests.sh` in the following manner.<br>
+  `./runtests.sh GAE_SDK_Location Project_Folder_location`<br>
+  e.g. `./runtests.sh User/someuser/appengine-java-sdk-1.9.27 User/someuser/TEAMMATES`, if the path to GAE_SDK_Location is `User/someuser/appengine-java-sdk-1.9.27` and path to Project_Folder_location is `User/someuser/TEAMMATES`.<br>
+  It's recommanded to use absolute path on Mac, and you could retrieve the absolute path by navigate to that folder and type `pwd` command.<br>
```
Comment (wkurniawan07) [inline devdocs/settingUp.md]: "recommended"

File: devdocs/settingUp.md [hunk 2]
```diff
@@ -179,7 +180,19 @@ this section.
   `runtests.bat  C:\appengine-java-sdk-1.9.27  C:\teammates -testnames component-tests,sequential-ui-tests,parallel-ui-tests`
 
 **On OS-X:**
-TBD
+* Build the project in Eclipse (`Project -> Clean`).
+* Start the dev server in Eclipse.
+* Open a Terminal window, navigate to the project folder and run the `runtests.sh` in the following manner.<br>
+  `./runtests.sh GAE_SDK_Location Project_Folder_location`<br>
+  e.g. `./runtests.sh User/someuser/appengine-java-sdk-1.9.27 User/someuser/TEAMMATES`, if the path to GAE_SDK_Location is `User/someuser/appengine-java-sdk-1.9.27` and path to Project_Folder_location is `User/someuser/TEAMMATES`.<br>
```
Comment (wkurniawan07) [inline devdocs/settingUp.md]: Tried it and it worked :+1: need to change `User/...` to `/User/...` though.
Also to make it more Mac-like, change `User` to `Users`.

File: devdocs/settingUp.md [hunk 3]
```diff
@@ -179,7 +180,19 @@ this section.
...
+  It's recommanded to use absolute path on Mac, and you could retrieve the absolute path by navigate to that folder and type `pwd` command.<br>
+  If the file is not an executable, change its permission by: `chmod +x runtests.sh`<br>
+  This will run the full test suite once and retry the failed tests several times.
+* The final result can be viewed by opening `[project folder]/testrunner/test-output/index.html`.
+* To run only certain `<test>` segments of the `testng.xml`, add the `-testnames`
+  option followed by the names of the `<test>` segments you want to run.<br>e.g.
+  `./runtests.sh User/someuser/appengine-java-sdk-1.9.27 User/someuser/TEAMMATES -testnames component-tests,sequential-ui-tests,parallel-ui-tests`
```
Comment (wkurniawan07) [inline devdocs/settingUp.md]: Here, too.

**Comment by wkurniawan07** [comment]
Few changes requested. Good job on taking this issue :+1:

File: devdocs/settingUp.md (hunk 2 above)
Comment (Gisonrg) [inline devdocs/settingUp.md]: Oh yea! How could I miss the 's' in 'Users' :stuck_out_tongue_closed_eyes:

File: devdocs/settingUp.md [hunk 4]
```diff
@@ -179,7 +180,19 @@ this section.
...
+* Open a Terminal window, navigate to the project folder and run the `runtests.sh` in the following manner.<br>
+  `./runtests.sh GAE_SDK_Location Project_Folder_location`<br>
+  e.g. `./runtests.sh /Users/someuser/appengine-java-sdk-1.9.27 /Users/someuser/TEAMMATES`, if the path to GAE_SDK_Location is `/Users/someuser/appengine-java-sdk-1.9.27` and path to Project_Folder_location is `/Users/someuser/TEAMMATES`.<br>
+  It's recommended to use absolute path on Mac, and you could retrieve the absolute path by navigate to that folder and type `pwd` command.<br>
+  If the file is not an executable, change its permission by: `chmod +x runtests.sh`<br>
+  This will run the full test suite once and retry the failed tests several times.
+* The final result can be viewed by opening `[project folder]/testrunner/test-output/index.html`.
+* To run only certain `<test>` segments of the `testng.xml`, add the `-testnames`
+  option followed by the names of the `<test>` segments you want to run.<br>e.g.
+  `./runtests.sh /Users/someuser/appengine-java-sdk-1.9.27 /Users/someuser/TEAMMATES -testnames component-tests,sequential-ui-tests,parallel-ui-tests`
```
Comment (wkurniawan07) [inline devdocs/settingUp.md]: Actually, can this be merged with the Windows instructions? Right now it's ~75% similar, and we can use `Do this on Windows (Mac: do this instead)` like what you've done in the previous section.

**Comment by Gisonrg** [comment]
@wkurniawan07 I merged two sections together. Actually I believe developers using Mac are smart enough to figure it out, LOL :laughing:

**Comment by wkurniawan07** [comment]
I would agree with that, but hey the issue is already there. Good job :+1:

**Pull request merged**